components/brave_wallet_ui/components/desktop/wallet-onboarding/restore/index.tsx

```diff
@@ -11,44 +11,90 @@ import {
   InputColumn,
   FormText
 } from './style'
-import { PasswordInput } from '../../../shared'
+import { PasswordInput, BackButton } from '../../../shared'
 import { NavButton } from '../../../extension'
 import locale from '../../../../constants/locale'
 import { Checkbox } from 'brave-ui'
 
 export interface Props {
-  onSubmit: () => void
-  onRecoveryPhraseChanged: (value: string) => void
-  onPasswordChanged: (value: string) => void
-  onConfirmPasswordChanged: (value: string) => void
-  recoveryPhrase: string
-  disabled: boolean
-  hasRestoreError?: boolean
-  hasPasswordError: boolean
-  hasConfirmPasswordError: boolean
+  toggleShowRestore: () => void
+  onRestore: (phrase: string, password: string) => void
+  hasRestoreError: boolean
 }
 
 function OnboardingRestore (props: Props) {
   const {
-    onSubmit,
-    onRecoveryPhraseChanged,
-    onConfirmPasswordChanged,
-    onPasswordChanged,
-    recoveryPhrase,
-    disabled,
-    hasRestoreError,
-    hasPasswordError,
-    hasConfirmPasswordError
+    onRestore,
+    toggleShowRestore,
+    hasRestoreError
   } = props
   const [showRecoveryPhrase, setShowRecoveryPhrase] = React.useState<boolean>(false)
+  const [password, setPassword] = React.useState<string>('')
+  const [confirmedPassword, setConfirmedPassword] = React.useState<string>('')
+  const [recoveryPhrase, setRecoveryPhrase] = React.useState<string>('')
 
-  const inputRecoveryPhrase = (event: React.ChangeEvent<HTMLInputElement>) => {
-    onRecoveryPhraseChanged(event.target.value)
+  const onBack = () => {
+    toggleShowRestore()
+    setRecoveryPhrase('')
   }
 
+  const onSubmitRestore = () => {
+    onRestore(recoveryPhrase, password)
+  }
+
+  const handlePasswordChanged = (value: string) => {
+    setPassword(value)
+  }
+
+  const handleConfirmPasswordChanged = (value: string) => {
+    setConfirmedPassword(value)
+  }
+
+  const handleRecoveryPhraseChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value
+    const removeBegginingWhiteSpace = value.trimStart()
+    const removedDoubleSpaces = removeBegginingWhiteSpace.replace(/ +(?= )/g, '')
+    const removedSpecialCharacters = removedDoubleSpaces.replace(/[^a-zA-Z ]/g, '')
+    if (recoveryPhrase.split(' ').length === 12) {
+      setRecoveryPhrase(removedSpecialCharacters.trimEnd())
+    } else {
+      setRecoveryPhrase(removedSpecialCharacters)
+    }
+  }
+
+  const isValidRecoveryPhrase = React.useMemo(() => {
+    const phrase = recoveryPhrase.split(' ')
+    if (phrase.length === 12 && phrase[11] !== '') {
```

Comment:
Can you help me understand what the `phrase[11] !== ''` part is for?

Reply:
Recovery phrases can be more than 12 words, 24 being a very common one. What do you think about the following check?

```js
recoveryPhrase.trim().split(/\s+/g).length >= 12
```

Reply:
`phrase[11] !== ''` is a check to make sure that the last string in the array is not empty.

All of this logic for recovery phrases is going to change in my next PR and will also allow for multiple lengths.

https://github.com/brave/brave-browser/issues/17532